test/e2e/kubevirt.go

```diff
@@ -214,6 +215,151 @@ passwd:
+		composeDiagnosticsDaemonSet = func(namespace, name, cmd, tool string) appsv1.DaemonSet {
+			ovnImage := os.Getenv("OVN_IMAGE")
+			if ovnImage == "" {
+				ovnImage = "localhost/ovn-daemonset-f:dev"
+			}
+			return appsv1.DaemonSet{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      name,
+					Namespace: namespace,
+				},
+				Spec: appsv1.DaemonSetSpec{
+					Selector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"app": name,
+						},
+					},
+					Template: v1.PodTemplateSpec{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      name,
+							Namespace: namespace,
+							Labels: map[string]string{
+								"app":  name,
+								"tool": tool,
+							},
+						},
+						Spec: v1.PodSpec{
+							Containers: []v1.Container{{
+								Name:    "ovn-kube",
+								Image:   ovnImage,
+								Command: []string{"bash", "-c"},
+								Args:    []string{cmd},
+								SecurityContext: &v1.SecurityContext{
+									Privileged: pointer.Bool(true),
+								},
+								VolumeMounts: []v1.VolumeMount{
+									{
+										Name:      "host-run-ovs",
+										MountPath: "/run/openvswitch",
+									},
+									{
+										Name:      "host-var-run-ovs",
+										MountPath: "/var/run/openvswitch",
+									},
+									{
+										Name:      "host",
+										MountPath: "/host",
+									},
+								},
+							}},
+							HostNetwork: true,
+							Volumes: []v1.Volume{
+								{
+									Name: "host-run-ovs",
+									VolumeSource: v1.VolumeSource{
+										HostPath: &v1.HostPathVolumeSource{
+											Path: "/run/openvswitch",
+										},
+									},
+								},
+								{
+									Name: "host-var-run-ovs",
+									VolumeSource: v1.VolumeSource{
+										HostPath: &v1.HostPathVolumeSource{
+											Path: "/var/run/openvswitch",
+										},
+									},
+								},
+								{
+									Name: "host",
+									VolumeSource: v1.VolumeSource{
+										HostPath: &v1.HostPathVolumeSource{
+											Path: "/",
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			}
+		}
+
+		runDaemonSets = func(daemonSets []appsv1.DaemonSet) error {
+			for _, daemonSet := range daemonSets {
+				_, err := fr.ClientSet.AppsV1().DaemonSets(daemonSet.Namespace).Create(context.Background(), &daemonSet, metav1.CreateOptions{})
+				if err != nil {
+					return err
+				}
+			}
+			for _, daemonSet := range daemonSets {
+				err := wait.PollUntilContextTimeout(context.Background(), time.Second, 15*time.Second, true /*immediate*/, func(ctx context.Context) (bool, error) {
+					daemonSet, err := fr.ClientSet.AppsV1().DaemonSets(daemonSet.Namespace).Get(ctx, daemonSet.Name, metav1.GetOptions{})
+					if err != nil {
+						return false, err
+					}
+					return daemonSet.Status.NumberAvailable == daemonSet.Status.NumberReady, nil
+				})
+				if err != nil {
+					return err
+				}
+			}
+			return nil
+		}
+
+		tcpdumpDaemonSet = func(port, nodePort string, ifaces []string) {
```

Comment:
instead of passing `port, nodePort string`, can't we just pass an optional filter string?